src/workerd/util/sqlite.c++

```diff
@@ -899,13 +903,19 @@ bool SqliteDatabase::isAuthorized(int actionCode,
     case SQLITE_DELETE:       /* Table Name      NULL            */
     case SQLITE_DROP_TABLE:   /* Table Name      NULL            */
     case SQLITE_INSERT:       /* Table Name      NULL            */
-    case SQLITE_ANALYZE:      /* Table Name      NULL            */
     case SQLITE_CREATE_VIEW:  /* View Name       NULL            */
     case SQLITE_DROP_VIEW:    /* View Name       NULL            */
     case SQLITE_REINDEX:      /* Index Name      NULL            */
       KJ_ASSERT(param2 == kj::none);
       return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1));
 
+    case SQLITE_ANALYZE: /* Table Name      NULL            */
+      KJ_ASSERT(param2 == kj::none);
+      // We allow all names (including names where isAllowedName() would return false) because a
+      // SQLite ANALYZE statement with no parameters will analyze all tables, including otherwise
```

Comment:
If I understand correctly, `ANALYZE` doesn't return any information to the caller, it just makes notes for future query planning. Is that right? Therefore, it certainly doesn't reveal nor modify the contents of hidden tables, therefore there's no need to restrict it? Might be worth spelling that out in the comment here.

Reply:
Your understanding is correct and I've updated the comment.

One non-obvious thing is that the notes that `ANALYZE` makes are available in a table that users *can modify*.  This means that customers can make the query planner do suboptimal things.  It's likely that the worst thing a customer could do is waste CPU, but it is a rather large surface that we open up by allowing `ANALYZE`.